src/main/java/dev/j3fftw/litexpansion/machine/Generator.java

```diff
@@ -0,0 +1,143 @@
+package dev.j3fftw.litexpansion.machine;
+
+import dev.j3fftw.litexpansion.Items;
+import io.github.thebusybiscuit.slimefun4.implementation.items.electric.generators.CoalGenerator;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.abstractItems.MachineFuel;
+import org.bukkit.Material;
+import org.bukkit.Tag;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class Generator extends CoalGenerator {
+
+    public Generator() {
+        super(Items.LITEXPANSION, Items.GENERATOR, RecipeType.ENHANCED_CRAFTING_TABLE, new ItemStack[] {
+            null, Items.RE_BATTERY, null,
+            null, Items.MACHINE_BLOCK, null,
+            null, new ItemStack(Material.FURNACE), null
+        });
+    }
+
+    @Nonnull
+    @Override
+    public String getInventoryTitle() {
+        return "&7Generator";
+    }
+
+    @Nonnull
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.FIRE_CHARGE);
+    }
+
+    @Override
+    public int getEnergyProduction() {
+        return 10;
+    }
+
+    @Override
+    protected void registerDefaultFuelTypes() {
+
+        registerFuel(new MachineFuel(100, new ItemStack(Material.LAVA_BUCKET)));
+        registerFuel(new MachineFuel(80, new ItemStack(Material.COAL_BLOCK)));
+        registerFuel(new MachineFuel(20, new ItemStack(Material.DRIED_KELP_BLOCK)));
+        registerFuel(new MachineFuel(12, new ItemStack(Material.BLAZE_ROD)));
+        registerFuel(new MachineFuel(8, new ItemStack(Material.COAL)));
+        registerFuel(new MachineFuel(8, new ItemStack(Material.CHARCOAL)));
+
+        for (Material mat : Tag.ITEMS_BOATS.getValues()) {
+            registerFuel(new MachineFuel(6, new ItemStack(mat)));
+        }
+
+        registerFuel(new MachineFuel(6, new ItemStack(Material.SCAFFOLDING)));
+
+        for (Material mat : Tag.LOGS.getValues()) {
+            registerFuel(new MachineFuel(2, new ItemStack(mat)));
+        }
+
+        for (Material mat : Tag.PLANKS.getValues()) {
+            registerFuel(new MachineFuel(1, new ItemStack(mat)));
+        }
```

Comment:
Most of this is super redundant, you are already extending CoalGenerator, so just invoke the super call instead of copy pasting everything...

Reply:
can i just extend the coal gen and then do the same stuff

Reply:
Yep, like I said: invoke the super call:
`super.registerDefaultFuelTypes()` and then all the ones from the super class will be added too